### PR TITLE
make: fix link to cri-o.service

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

 /kind cleanup

#### What this PR does / why we need it:
when installing with package managers, we leak a broken symlink, as we link to the in-dir crio.service, instead of the one recently installed
#### Which issue(s) this PR fixes:
a 1.17 version if this pr will fix https://github.com/cri-o/cri-o/issues/3879

#### Does this PR introduce a user-facing change?

```release-note
fix broken symlink of systemd unit file
```

### CODE REVIEW

**Comment by haircommander** [comment]
rebased to pick up test fixes

/retest

**Comment by TomSweeneyRedHat** [comment]
@saschagrunert looks like the CI might not be happy making the link the way that you want to.

**Comment by TomSweeneyRedHat** [comment]
oops, @haircommander, too many PRs in the inbox this morn.

**Comment by haircommander** [comment]
/retest

let's see if ci is forgiving :) (welcome back @TomSweeneyRedHat )

**Comment by haircommander** [comment]
test failures actually seem legit here

File: Makefile [hunk 1]
```diff
@@ -468,7 +468,7 @@ install.completions:
 
 install.systemd:
 	install ${SELINUXOPT} -D -m 644 contrib/systemd/crio.service $(PREFIX)/lib/systemd/system/crio.service
-	ln -sf crio.service $(PREFIX)/lib/systemd/system/cri-o.service
+	ln -sf $(PREFIX)/lib/systemd/system/crio.service $(PREFIX)/lib/systemd/system/cri-o.service
```
Comment (jayaddison) [inline Makefile:471]: @haircommander Perhaps requesting `ln` create a relative link might work?  At a guess, the target path (including BUILDROOT reference) is being expanded before the link is written to the filesystem, if that makes sense -- and it succeeds despite no such target existing, due to the 'force' flag.

Comment (haircommander) [reply Makefile:471]: thanks for the suggestion! I looked into a bit more, and it turns out symlinks between absolute paths in rpms are just not allowed. I am looking into a new solution now, which may not involve these changes...

**Comment by haircommander** [comment]
this change is now primarily cosmetic: the real change is happening in the packaging solutions (and looks very similar to the change in cri-o.service)

File: Makefile (hunk 1 above)
Comment (jayaddison) [reply Makefile]: No probs!  Thanks for investigating further

**Comment by haircommander** [comment]
@mrunalp @umohnani8 @saschagrunert @giuseppe @lsm5 PTAL

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by haircommander** [comment]
@mrunalp @umohnani8 @kolyshkin @saschagrunert @fidencio ptal

File: contrib/test/ci/cri-o.spec [hunk 2]
```diff
@@ -135,9 +135,11 @@ export GOPATH=%{buildroot}/%{gopath}:$(pwd)/Godeps/_workspace:%{gopath}
 %endif
 
 %post
+ln -sf %{_unitdir}/%{service_name}.service {_unitdir}/%{repo}.service
```
Comment (fidencio) [inline contrib/test/ci/cri-o.spec:138]: There's a typo here:
`{_unitdir}/%{repo}.service` should be `%{_unitdir}/%{repo}.service`

Comment (haircommander) [reply contrib/test/ci/cri-o.spec]: good catch!

**Review by saschagrunert — APPROVED** [review]
LGTM

**Review by lsm5 — APPROVED** [review]
LGTM

**Comment by jayaddison** [comment]
At a guess, merging/rebasing to an updated branch/commit that includes 2428fcc3ea38e6f8938062a396acbd7237154368 might help the tests pass here?

**Comment by haircommander** [comment]
> At a guess, merging/rebasing to an updated branch/commit that includes [2428fcc](https://github.com/cri-o/cri-o/commit/2428fcc3ea38e6f8938062a396acbd7237154368) might help the tests pass here?

good guess, but these e2es are slightly different, and are failing because of general infrastructure stability :)

/retest

**Comment by jayaddison** [comment]
No probs - the `In-tree Volumes` test in a previous build log seemed to be where things were [going wonky](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/cri-o_cri-o/3890/pull-ci-cri-o-cri-o-master-e2e-gcp/1333831814095048704#1:build-log.txt%3A2632) and hitting GCP rate limits; I figured the change (which includes disabling that test) might affect that overall stability.

**Comment by haircommander** [comment]
thanks for looking into it :slightly_smiling_face:

**Pull request merged**